### PR TITLE
Jets

### PR DESCRIPTION
**Pull request by martyr-binbex**

### CODE REVIEW

File: lib/zink/zink.hoon [hunk 1]
```diff
@@ -206,6 +210,68 @@
...
+    =^  sub=body  app  ^$(f head)
+    ?@  sub  `app
+    =^  arg=body  app  ^$(s sub^s, f sam.next)
+    ?@  arg  `app
+    (both mjet arg)^app
+  ::
+  ++  run-jet
+    |=  [arm=@tas sam=*]
+    ^-  book
+    ?+  arm  `app
```
Comment (tacryt-socryp) [inline lib/zink/zink.hoon:247]: I wonder if we can make this more elegant by doing something a bit more raw. I need to think about it. It's clear as is, and obviously works though.

File: lib/zink/zink.hoon [hunk 2]
```diff
@@ -206,6 +210,68 @@
...
+  ::
+  ++  match-jet
+    |=  [head=* next=*]
+    ^-  [(unit [@tas *]) appendix]
+    ?:  (lth bud 1)  `app
+    =.  bud  (sub bud 1)
+    ?.  ?=([%9 arm-axis=@ %0 core-axis=@] head)  `app
+    ?.  ?=([%9 %2 %10 [%6 sam=*] %0 %2] next)  `app
+    =/  mjet  (~(get by jets) arm-axis.head)
+    ?@  mjet  `app
```
Comment (tacryt-socryp) [inline lib/zink/zink.hoon:237]: Should be `?~` and can be defined inline:

?~  mjet=...

File: lib/zink/zink.hoon [hunk 3]
```diff
@@ -206,6 +210,68 @@
...
+    =^  sub=body  app  ^$(f head)
+    ?@  sub  `app
```
Comment (tacryt-socryp) [inline lib/zink/zink.hoon:239]: Should be ?~

File: lib/zink/zink.hoon [hunk 4]
```diff
@@ -206,6 +210,68 @@
...
+    =^  arg=body  app  ^$(s sub^s, f sam.next)
+    ?@  arg  `app
```
Comment (tacryt-socryp) [inline lib/zink/zink.hoon:241]: Should be ?~

File: lib/zink/zink.hoon [hunk 5]
```diff
@@ -206,6 +210,68 @@
...
+    ^-  [(unit [@tas *]) appendix]
+    ?:  (lth bud 1)  `app
+    =.  bud  (sub bud 1)
+    ?.  ?=([%9 arm-axis=@ %0 core-axis=@] head)  `app
+    ?.  ?=([%9 %2 %10 [%6 sam=*] %0 %2] next)  `app
+    =/  mjet  (~(get by jets) arm-axis.head)
+    ?@  mjet  `app
+    =^  sub=body  app  ^$(f head)
+    ?@  sub  `app
+    =^  arg=body  app  ^$(s sub^s, f sam.next)
```
Comment (tacryt-socryp) [inline lib/zink/zink.hoon:240]: Style, put the ^$ on the next line

File: lib/zink/zink.hoon [hunk 6]
```diff
@@ -206,6 +210,68 @@
...
+  ++  match-jet
+    |=  [head=* next=*]
+    ^-  [(unit [@tas *]) appendix]
+    ?:  (lth bud 1)  `app
+    =.  bud  (sub bud 1)
+    ?.  ?=([%9 arm-axis=@ %0 core-axis=@] head)  `app
+    ?.  ?=([%9 %2 %10 [%6 sam=*] %0 %2] next)  `app
+    =/  mjet  (~(get by jets) arm-axis.head)
+    ?@  mjet  `app
+    =^  sub=body  app  ^$(f head)
```
Comment (tacryt-socryp) [inline lib/zink/zink.hoon:238]: Style, put the ^$ on next line

File: lib/zink/zink.hoon [hunk 7]
```diff
@@ -206,6 +210,68 @@
...
+  :: If we match this then look up JET-AXIS in the jet map to see if we're
+  :: calling a jetted arm.
+  ::
+  :: Note that this arm should only be called on an 8
+  :: TODO Figure out what CORE-AXIS should be
+  ++  jet
+    |=  [head=* next=*]
+    ^-  book
+    =^  mj  app  (match-jet head next)
+    ?@  mj  `app
```
Comment (tacryt-socryp) [inline lib/zink/zink.hoon:226]: Use ?~ for matching ~

File: sur/zink.hoon [hunk 8]
```diff
@@ -37,7 +37,16 @@
 +$  hints  (mip phash phash cairo-hint)
 ::  map of a noun's merkle children. root -> [left right]
 +$  merk-tree  (map phash [phash phash])
-::
+::  map from axis to jet name
++$  jetmap  (map @ @tas)
```
Comment (tacryt-socryp) [inline sur/zink.hoon:41]: Please incorporate a gas cost function in this map that takes in the same input as the target jet and is then run to determine cost.

**Review by tacryt-socryp — CHANGES_REQUESTED** [review]

File: lib/zink/zink.hoon (hunk 7 above)
Comment (martyr-binbex) [reply lib/zink/zink.hoon:226]: That requires `?:  ?~(mj) app` which is more verbose

Comment (tacryt-socryp) [reply lib/zink/zink.hoon:226]: No, it doesn't. You can drop it in place of your ?@

File: lib/zink/zink.hoon (hunk 4 above)
Comment (martyr-binbex) [reply lib/zink/zink.hoon:241]: I realized this line can be deleted since `both` will return ~ if either unit is ~

**Review by tacryt-socryp — APPROVED** [review]
LGTM

**Pull request merged**